docs/guides/using-oidc-plugin.md

```diff
@@ -0,0 +1,145 @@
+# Using OIDC plugin
+
+Kong Enterprise's OIDC plugin can authenticate requests using OpenID Connect protocol.
+This guide shows a basic example of how to setup the OIDC plugin using
+the Ingress Controller.
+
+> Note: This works only with Enterprise version of Kong.
+
+## Installation
+
+Please follow the [deployment](../deployment/k4k8s-enterprise.md) documentation
+to install enterprise version of Kong Ingress Controller.
+
+## Testing Connectivity to Kong
+
+This guide assumes that the `PROXY_IP` environment variable is
+set to contain the IP address or URL pointing to Kong.
+Please follow one of the
+[deployment guides](../deployment) to configure this environment variable.
+
+If everything is setup correctly, making a request to Kong should return
+HTTP 404 Not Found.
+
+```bash
+$ curl -i $PROXY_IP
+HTTP/1.1 404 Not Found
+Date: Fri, 21 Jun 2019 17:01:07 GMT
+Content-Type: application/json; charset=utf-8
+Connection: keep-alive
+Content-Length: 48
+Server: kong/1.2.1
+
+{"message":"no Route matched with those values"}
+```
+
+This is expected as Kong does not yet know how to proxy the request.
+
+## Setup a Sample Service
+
+For the purpose of this guide, we will setup an [httpbin](https://httpbin.org)
+service in the cluster and proxy it.
+
+```bash
+kubectl apply -f https://bit.ly/k8s-httpbin
+service/httpbin created
+deployment.apps/httpbin created
+```
+
+Create an Ingress rule to proxy the httpbin service we just created:
+
+```bash
+$ echo '
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: demo
+spec:
+  rules:
+  - host: 34.82.120.8.xip.io
+    http:
+      paths:
+      - path: /
+        backend:
+          serviceName: httpbin
+          servicePort: 80
+' | kubectl apply -f -
+ingress.extensions/demo created
+```
+
+We are using `34.82.120.8.xip.io` as our host, you can use any domain name
```

Comment:
I'd recommend replacing the actual GCP IP here with something from the reserved documentation range, 192.0.2.0/24, e.g. `192.0.2.8`. We may no longer be using this particular real IP, but it will likely be assigned to someone else in the future. Ditto rest of the document, `sed -i -e "s/34.82.120.8/192.0.2.8/g" docs/guides/using-oidc-plugin.md` should get the lot.